lib/connection_string.js

```diff
@@ -669,7 +674,7 @@ function parseConnectionString(uri, options, callback) {
         parsedHost.port = null;
       }
 
-      if (Number.isNaN(parsedHost.port)) {
+      if (Number.isNaN(+parsedHost.port)) {
```

Comment:
I've never really used `NaN` type guards, what's the idea of `Number.isNaN(+` here?

Reply:
`+` in front of a variable coerces it to be a number even `+'NaN' === NaN`, typescript complains cus port is a string.